### ISSUE TITLE
Cookie not decoded

### ISSUE THREAD

**Issue by dype35** [nojimage/cakephp-remember-me#8]
hello
The cookie is saved but not decoded.

The site in https
cakephp 3.6.9 (but same problem with 3.6.7)

The config : 
```
$this->loadComponent('Auth', [
            'authenticate' => [
                'RememberMe.Cookie' => [
                    'userModel' => 'Users',
                    'fields' => ['username' => 'user_login'],
                    'always' => true,
                    //'inputKey' => 'remember_me',

                    'cookie' => [
                        //'name' => 'rememberMe',
                        //'expires' => '+10 days',
                        'secure' => true,
                        'httpOnly' => false,
                    ],

                ],
                'Form' =>[
                    'fields' => ['username' => 'user_login','password' => 'user_pass'],
...
```
When the user is loged the fist time:
- there is an entry in the databse
- a cookie is created

When the user try to reconnect after some time : 
 - the cookie is send to the application
 - **the cookie is not decoded**
so the user must loggin again

I add some debug in your code (but found no solution to correct it) 
1st debug  in function checkFields
```
$cookie = $this->getCookie($request);
Log::write('debug',$cookie);
```
I can see : 
 ```
2018-07-30 21:55:57 Debug: Q2FrZQ==.MGE0YTc3MzJjMzhjMzM1ODg5NjU0M2U5NDZjYzU1YWY5ODczOGRkMmEzZTA0NGYzYjk3OGExMzgzODU0YTFkOfZ/eLunwTFYUBZ7SqIPdBr0d
EVIatGD6l8lOLrftHbE+JcgikbUoAQPBQscnsKm21p3FT3TgHiUWxP96nZAJy8=
```
2nd debug in function checkFields again
```
$decoded = $this->decodeCookie($cookie);
Log::write('debug',$decoded);
```
$decoded variable is empty : 
`2018-07-30 21:55:57 Debug: `

Thanks for your help
F.

**Comment by nojimage** [comment]
Are you using EncryptedCookieMiddleware?
If EncryptedCookieMiddleware is enabled when login and you disabled EncryptedCookieMiddleware after login, please try enabled EncryptedCookieMiddleware again.

**Comment by dype35** [comment]
No I do not use EncryptedCookieMiddlewar. How can I check that ? I have to
activate it ?

But this application comes from an old one develop witch cakephp 2.x and I
used the cookie module of cake 2.x,
Perhaps something is still present ?

**Comment by nojimage** [comment]
Please check `src/Application.php` to see if middleware is active.

See also:

- Using Middleware https://book.cakephp.org/3.0/en/controllers/middleware.html#using-middleware
- Encrypted Cookie Middleware https://book.cakephp.org/3.0/en/controllers/middleware.html#encrypted-cookie-middleware

**Comment by nojimage** [comment]
I also tried enabling EncryptedCookieMiddleware.
I saw that the cookie was not decoded, too.
It's seems this plugins cookie was include binary, then decode failure.

I'll fix it.

**Comment by nojimage** [comment]
fixes it!, [v2.2.0](https://github.com/nojimage/cakephp-remember-me/releases/tag/v2.2.0) released!

thank you for reporting. 😄
